### PR TITLE
Change: Message.time was float, becomes datetime

### PR DESCRIPTION
**Pull request by hoh**
Change the type if BaseMessage.time to a datetime. This allows easier manipulation of the field, and parsing from strings as well as floats.

New APIs introduced in PyAleph 0.5.0 store the message time as timestampz and return it as ISO strings. This changes is compatible with both the old and new APIs.

### CODE REVIEW

**Review by BjrInt — APPROVED** [review]
Looks good to me, but:
Shouldn't the pydantic model also explicitly include `Float` for clarity?

**Comment by hoh** [comment]
This would send datetimes as strings, which is likely not compatible with the MongoDB API - we should wait for the PostgreSQL based version of MyAleph to release this.

**Comment by hoh** [comment]
Moved as draft, all nodes should be upgraded to use Postgres beforehand.

**Comment by MHHukiewitz** [comment]
> Moved as draft, all nodes should be upgraded to use Postgres beforehand.

Nodes are using Postgres now, should we continue pursing this PR?

**Comment by hoh** [comment]
Add a @validator to convert float to datetime

**Comment by MHHukiewitz** [comment]
@hoh it's ready

File: aleph_message/models/__init__.py [hunk 1]
```diff
@@ -265,6 +266,13 @@ def check_confirmed(cls, v, values):
             raise ValueError("Message cannot be 'confirmed' without 'confirmations'")
         return v
 
+    @validator("time")
+    def check_time(cls, v, values):
```
Comment (hoh) [inline aleph_message/models/__init__.py:270]:
```suggestion
    def convert_float_to_datetime(cls, v, values):
```

**Pull request merged**